Yoast/wordpress-seo
# Remove Indexable Meta model and replace it with Indexable Extension

## Summary

This PR can be summarized in the following changelog entry:

* No user facing changes

## Relevant technical choices:

* Optimizes the opengraph and twitter attributes of the Indexable by adding them to the model directly.
* Removes the `Indexable_Meta` model.
* Adds a `permalink_hash` column which is indexed to allow indexed queries on permalink.
* Adds the `Indexable_Extension` abstract class which is intended to be inherited by AddOns to serve as a basis for their own custom models to store their data.

## Test instructions


This PR can be tested by following these steps:

* When enabling the `YOAST_SEO_INDEXABLES` feature flag entries should be added to the indexable table when saving posts.
* Use `wp shell` to run `\Yoast\WP\Free\Models\Indexable::find_by_url`, it should return indexables with an exact url match.
* Create a `Indexable_Video_Extension` class and table with some relevant attributes. Register this class as a extension and make sure it's retrievable using `$indexable->get_extension( 'news' )`

## UI changes

## Quality assurance


Fixes #13078
Fixes #11233
